packages/@webex/plugin-cc/src/services/task/TaskManager.ts

```diff
@@ -0,0 +1,119 @@
+import EventEmitter from 'events';
+import {ICall, LINE_EVENTS} from '@webex/calling';
+import {WebSocketManager} from '../core/websocket/WebSocketManager';
+import routingContact from './contact';
+import WebCallingService from '../WebCallingService';
+import {ITask, TASK_EVENTS, TaskId} from './types';
+import {CC_EVENTS} from '../config/types';
+import {LoginOption} from '../../types';
+import Task from '.';
+
+export default class TaskManager extends EventEmitter {
+  private call: ICall;
+  private contact: ReturnType<typeof routingContact>;
+  private taskCollection: Record<TaskId, ITask>;
+  private webCallingService: WebCallingService;
+  private webSocketManager: WebSocketManager;
+  private static taskManager;
+  public currentTask: ITask;
+
+  /**
+   * @param contact - Routing Contact layer. Talks to AQMReq layer to convert events to promises
+   * @param webCallingService - Webrtc Service Layer
+   * @param webSocketManager - Websocket Manager to maintain websocket connection and keepalives
+   */
+  constructor(
+    contact: ReturnType<typeof routingContact>,
+    webCallingService: WebCallingService,
+    webSocketManager: WebSocketManager
+  ) {
+    super();
+    this.contact = contact;
+    this.taskCollection = {};
+    this.webCallingService = webCallingService;
+    this.webSocketManager = webSocketManager;
+    this.registerTaskListeners();
+    this.registerIncomingCallEvent();
+  }
+
+  private handleIncomingWebCall = (call: ICall) => {
+    if (this.currentTask) {
+      this.emit(TASK_EVENTS.TASK_INCOMING, this.currentTask);
+      // this.currentTask.webCallMap[this.currentTask.data.interactionId] = call.getCallId();
```

Comment:
Didn't we discuss about webCallingService containing this map, an dnot the currentTask? Is there a ticket to address this?

Reply:
https://jira-eng-gpk2.cisco.com/jira/browse/SPARK-589789
Added a point in this JIRA to address the map